onnxruntime/python/tools/quantization/shape_inference.py

```diff
@@ -63,6 +65,14 @@ def quant_pre_process(
         external_data_location: The file location to save the external file
         external_data_size_threshold: The size threshold for external data
     """
+
+    # input_model_path is deprecated, this is for backward compatibility
+    # when input_model is provided, we will ignore input_model_path.
+    if input_model is None:
+        input_model = input_model_path
+
```

Comment:
assert input_model is not None, "either input_model or input_model_path shall be specified"